### ISSUE TITLE
memory leak with Docker.qcow2

### ISSUE THREAD

**Issue by ilg-ul** [docker/for-mac#2279]
### Expected behavior

- create as many Docker images as the disk space allows

### Actual behavior

- after a while, the `Docker.qcow2` reaches a max size and no more Docker builds are possible

### Information

```
Docker for Mac: version: 17.09.0-ce-mac35 (69202b202f497d4b6e627c3370781b9e4b51ec78)
macOS: version 10.13.1 (build: 17B1002)
logs: /tmp/116AB40C-26F8-4FB2-80F7-8EB9648336BD/20171130-125655.tar.gz
[OK]     db.git
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     db
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     slirp
[OK]     osxfs
[OK]     moby-console
[ERROR]  logs
         logs check failed with: (Failure
  "exec: /usr/bin/log show --style syslog --last 1d | grep -i docker > \"/tmp/116AB40C-26F8-4FB2-80F7-8EB9648336BD/20171130-125655/docker-system-os.log\": exit 1")
[OK]     docker-cli
[OK]     menubar
[OK]     disk
```

### Steps to reproduce the behavior

- run a long sequence of very large Docker builds
- from time to time remove unused images
- depending on image size, after a while Docker will run out of space, even if `docker images` shows only a few images

Once this happens, the only solution is the Preferences -> Reset button, that removes all images.

I have no idea why you need to keep all images in a single file, and do not have separate files for each image.

I noticed a change in 7.11, replacing `.qcow2` by `.raw` files, which improves performances, but if space management is the same, I'm afraid the memory leak will affect the raw images too.

**Comment by guillaumerose** [comment]
In the next edge version (next week if everything is ok), we will introduce a slider in the UI that allow to set the Qcow image size.

**Comment by ilg-ul** [comment]
yes, but this does not solve the memory leak, it'll simply consume more an more space on disk.

is the problem related to a memory leak bug, or this is the expected behaviour, to never free the space, even if the images are removed?
